api/api.go

```diff
@@ -61,10 +62,11 @@ type API struct {
 	port    int
 	version string
 	srv     *http.Server
+	tls     *config.TLSConfig
 }
 
 // NewAPI create a new API object
-func NewAPI(store *event.Store, drivers *driver.Drivers, port int) *API {
+func NewAPI(store *event.Store, drivers *driver.Drivers, port int, tls *config.TLSConfig) *API {
```

Comment:
Nit: 3 parameters is a load, and now with 4 it might be a good time to make a dedicated config struct for the API to pass in. Where can configure the options optionally and the function can handle default values.

Like

```
func NewAPI(cfg myStruct) *API {
  var api &API{}
  if cfg.Store == nil {
    api.Store = event.NewStore()
  }
}

Reply:
Makes sense, will put these options in a new struct!